client/signup/steps/domains/index.jsx

```diff
@@ -380,6 +380,11 @@ class DomainsStep extends Component {
 			Object.assign( { domainItem }, useThemeHeadstartItem )
 		);
 
+		// Skip the plans step if we are in the `domain-transfer` flow.
+		if ( 'domain-transfer' === this.props.flowName ) {
```

Comment:
Similarly, something feels odd here. Is this condition even necessary?

Also, I recall "skipping a step" in this framework was more complicated than just submitting the step from a previous step. I feel this needs more testing. What happens if you go into `/start/domain-transfer/plans` directly or in any sort of clickthrough fashion ("go back" from browser, etc). I recall we needed to create a different/sibling flow and redirect to it on a couple of occasions (I can find some links if it comes to it).

Reply:
I did recall the same thing, also having used `useBranchSteps` before. However, in this case, I took inspiration from `maybeExcludeEmailsStep`: https://github.com/Automattic/wp-calypso/blob/90882ceb39d4284a278d24af3d85a7e4428de6b5/client/lib/signup/step-actions/index.js#L1053-L1086

This function also includes this two extra lines which I missed adding here:
```
	recordExcludeStepEvent( stepName, siteUrl );

	flows.excludeStep( stepName );
```

I'll dig into what these do and add them if necessary.